File: java/0217ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

class Solution {
  public boolean containsDuplicate(int[] nums) {
    Set<Integer> map = new HashSet<Integer>();

    for (int i = 0; i < nums.length; i++) {
      if (map.contains(nums[i])) {
        return true;
      } else {
        map.add(nums[i]);
      }
    }

    return false;
  }  
}

File: js/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
  const elements = new Set()
  for (let i = 0; i < nums.length; i++) {
    if (elements.has(nums[i])) return true
    elements.add(nums[i])
  }
  return false
};